File: 0525-contiguous-array/0525-contiguous-array.py
class Solution:
    def findMaxLength(self, nums: List[int]) -> int:
        mpp={0:-1}
        pre=0; res=0
        for j in range(len(nums)):
            if nums[j]==0:
                nums[j]=-1
        #print(nums)
        for i in range(len(nums)):
            pre+=nums[i]
            #mpp[pre]=i
            if pre in mpp:
                res=max(res,i-mpp[pre])
            else:mpp[pre]=i
        return res

File: 0525-contiguous-array/0525-contiguous-array.cpp
class Solution {
public:
    int findMaxLength(vector<int>& nums) {

        int n=nums.size();

        map<int,int>m;
        int ans=0;
        int pre=0;


        m[pre]=-1;

        for(int i=0;i<n;i++)
        {
            if(nums[i]==0)
            {
                nums[i]=-1;

            }
        }

        for(int i=0;i<n;i++)
        {
            pre+=nums[i];

            if(m.find(pre)!=m.end()){
                int val=m[pre];
                ans=max(ans,i-val);
            }
            else
            {
                m[pre]=i;
            }
        }

        return ans;
        
    }
};